### ISSUE TITLE
Why same optimizer is being used to optimize both generators in cycle GAN ?

### ISSUE THREAD

**Issue by asjad18** [eriklindernoren/PyTorch-GAN#108]
optimizer_G = torch.optim.Adam(
    itertools.chain(G_AB.parameters(), G_BA.parameters()), lr=opt.lr, betas=(opt.b1, opt.b2)
)

I am unable to understand this. Why shouldn't we use different optimizer for both generators ?

**Comment by AnimationFan** [comment]
just see the paper , the cycle consistentency loss has consist the loss of the two generator , the purpose of the loss it's to make the generator perform better at the same time, so it's right to optimize both generator
